src/libs/NetworkConnection.js

```diff
@@ -86,6 +109,19 @@ function onReconnect(callback) {
     reconnectionCallbacks.push(callback);
 }
 
+function recheckNetworkConnection() {
+    if (hasPendingNetworkCheck) {
+        return;
+    }
+
+    Log.info('[NetworkConnection] recheck NetInfo');
+    hasPendingNetworkCheck = true;
+    unsubscribeFromNetInfo();
```

Comment:
@kidroca Do you recall why you chose to unsubscribe and resubscribe to NetInfo rather than using [NetInfo.fetch](https://github.com/react-native-netinfo/react-native-netinfo#fetch) to recheck the network connection?

Reply:
Oh, looks like NetInfo will only trigger the recheck if you call `NetInfo.fetch` with a specific interface: https://github.com/react-native-netinfo/react-native-netinfo/blob/f05f7dd6e829135d907506e9270e729066b6e5d1/src/internal/state.ts#L100-L106

Reply:
> Oh, looks like NetInfo will only trigger the recheck if you call `NetInfo.fetch` with a specific interface

Yes, there isn't an exposed way to force a recheck, it might not be necessary, though we decided to do it if any request is taking unusually long time (more than 10-15 sec)

What triggers the recheck is actually the call to `NetInfo.configure`

Reply:
> Yes, there isn't an exposed way to force a recheck

Yep, I created a PR for react-native-netinfo to add this and clean up this implementation a bit: https://github.com/react-native-netinfo/react-native-netinfo/pull/594